transforms/dec128.js

```diff
@@ -0,0 +1,76 @@
+const DECIMAL_LITERAL = "DecimalLiteral";
+
+const opToName = {
+  '+': 'add',
+  '*': 'mul',
+  '-': 'sub',
+  '/': 'div',
+};
+
+const includedOps = new Map(Object.entries(opToName));
+
+const replaceWithDecimalExpression = (t, knownDecimalNodes) => (path) => {
+
+  const { left, right, operator } = path.node;
+
+  const leftIsDecimal = knownDecimalNodes.has(left);
+  const rightIsDecimal = knownDecimalNodes.has(right);
+
+  if (!leftIsDecimal && !rightIsDecimal) {
+    return;
+  }
+
+  if (leftIsDecimal !== rightIsDecimal) {
+    throw path.buildCodeFrameError(new SyntaxError('Mixed numeric types are not allowed.'));
+  }
+
+  if (!includedOps.has(operator)) {
+    throw path.buildCodeFrameError(new SyntaxError(`${operator} is not currently supported.`));
+  }
+
+  const member = t.memberExpression(
+    left,
+    t.identifier(opToName[operator])
+  );
+
+  const newNode = t.callExpression(
+    member,
+    [right]
+  );
+
+  knownDecimalNodes.add(newNode);
+
+  path.replaceWith(newNode);
+  path.skip();
+};
+
+const replaceWithDecimal = (t) => (path) => {
+
+  const num = t.stringLiteral(path.node.value);
+  const callee = t.identifier('Decimal');
+
+  path.replaceWith(
+    t.callExpression(callee, [num])
+  );
+};
+
+const addToDecimalNodes = (t, knownDecimalNodes) => (path) => {
+  knownDecimalNodes.add(path.node);
```

Comment:
This should check if `path.get("callee").isIdentifier({ name: "Decimal" })`, otherwise it compiles code like `foo() + bar()`.

Reply:
Lol, I just accidentally deleted it for some reason: https://github.com/sarahghp/decimal-playground/commit/0fb488ac993d7963de4995d85d33bcd582a94c82#diff-87f19a9f32c6381a9bdb4ca36647714e7f4f917bd7b58d58c98cf8cc4ec03d65L56